### PR TITLE
49 refactor the userindb

### PR DESCRIPTION
**Pull request by Maximelego**
Refactored the UserInDB to separate accounts and authentication from profiles.
- Added CRUD operations on accounts
- Fixed CRUD operations on profiles.
- Renamed Users into Profiles.
- Refactored the permission-checking method to take into account the academic year and to use the account instead of profile.
- Added basic JSON models for database initialisation for accounts.
- Fixed the Profiles JSON models.

Added several QOL features : 
- Added abstract models (pydantic and tortoise) for academic years as well as validators.
- Added tags in routers to make OpenAPI documentation easier.
- Added cleaning utilities to clean __pycache__ folders when using the cleaning script. 
- Fixed Typing of tokens.

**This PR has breaking features for the frontend.**
Several operations such as **user creation and authentication** will probably not work anymore.

### CODE REVIEW

File: app/routes/tags.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+"""
+This module contains the Tag class.
+"""
+class Tag():
```
Comment (desaleo) [inline app/routes/tags.py:4]: useless parenthesis

File: app/main.py [hunk 2]
```diff
@@ -10,20 +10,15 @@
 from fastapi.middleware.cors import CORSMiddleware
 from starlette.responses import RedirectResponse
 
-from app.routes import auth, user
+from app.routes import account, auth, profile
 
 from app.utils.databases.db import startup_databases
 
 # Array for the routes descriptions and names.
 tags_metadata: list[dict[str, str]] = [
```
Comment (desaleo) [inline app/main.py:18]: could this be replaced with a TypedDict?

File: app/models/pydantic/ProfileModel.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    firstname:  Name
+    lastname:   Name
+    mail:       Mail
+    account_id: Optional[int] = None
+
+    class Config:
+        """
+        Config class used to allow the model to be created from a dictionary.
+        """
+        from_attributes = True
```
Comment (desaleo) [inline app/models/pydantic/ProfileModel.py:42]: missing type

File: app/routes/tags.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    """
+
+    name: str
+    description: str
+
+    def __init__(self, name: str, description: str):
+        self.name = name
+        self.description = description
+
+    def export(self) -> dict[str, str]:
```
Comment (desaleo) [inline app/routes/tags.py:17]: could this be replaced by a TypedDict?

**Review by desaleo — APPROVED** [review]
LGTM

File: app/models/pydantic/validator.py [hunk 5]
```diff
@@ -44,10 +45,24 @@ def is_name(string: str) -> str:
...
 
 Login = Annotated[str, AfterValidator(is_login)]
+
+def is_academic_year(years: tuple[int, int]) -> tuple[int, int]:
+    """
+    We check that the tuple contains 2 values and that the second element is 
+    equal to the first one + 1.
+    """
+    if len(years) != 2:
+        raise ValueError("Le tuple doit contenir exactement deux éléments.")
```
Comment (HugoBien) [inline app/models/pydantic/validator.py:63]: C'est normal qu'on ne renvoit pas une requete HTTP ?

Comment (Maximelego) [reply app/models/pydantic/validator.py:63]: Tecnhically yes, but actually no.
ValueErrors are like assertions, they return 422 unprocessable entities.

**Review by HugoBien — APPROVED** [review]

**Pull request merged**